streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -552,8 +552,14 @@ private void handleTasksWithStateUpdater(final Map<TaskId, Set<TopicPartition>>
 
     private void handleTasksPendingInitialization() {
         // All tasks pending initialization are not part of the usual bookkeeping
+
+        final Set<Task> tasksToCloseDirty = new HashSet<>();
+
         for (final Task task : tasks.drainPendingTasksToInit()) {
-            closeTaskClean(task, Collections.emptySet(), Collections.emptyMap());
+            closeTaskClean(task, tasksToCloseDirty, new HashMap<>());
+        }
+        for (final Task task : tasksToCloseDirty) {
+            closeTaskDirty(task, false);
```

Comment:
Not 100% sure if we need to pass `true` or `false` here?

Reply:
You should pass `false`. With the state updater newly created or revived tasks are not added to the task registry immediately but only when they are ready for running.